### PR TITLE
Use git rev-parse to find git directory

### PR DESCRIPTION
**Pull request by selavy**
The gitrevision.h header is updated when a new commit is made. New
commits are detected by adding a cmake dependency on the logs/HEAD file
in the .git directory. Unfortunately, submodules do not have a .git
directory (it is a file with the path to the parent's git directory).
So instead ask git where the .git directory is via git rev-parse.

The determination of the .git directory is split from the file
generation command because git rev-parse's behavior is not consistent
across versions of git and between being called from a submodule or a
top-level repository. To get around this, we want to do a cross-platform
equivalent to `readlink -f $(git rev-parse --git-path logs/HEAD)`, which
in cmake parlance is get_filename_component(<PATH> ABSOLUTE). This needs
to be done from $PROJECT_SOURCE_DIR for consistency. Additionally,
add_custom_command() is run at build time rather than cmake configuration
time, which is when add_library() is done. I'm not sure why, but cmake
appears requires add_custom_command() to be in the same CMakeLists.txt
file as the add_library() call that has gitrevision.h as a source
dependency.

In short, with this change cgreen can now be configured as a top-level
repo with an in-source or out-of-source build as well as a submodule
again with either an in-source or out-of-source build.

### CODE REVIEW

**Comment by selavy** [comment]
For https://github.com/cgreen-devs/cgreen/issues/199

**Comment by souryogurt** [comment]
It works for me! Thank you for your contribution!
fedora 31
git 2.23.0
